src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimWithDocumentManagementTest.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.hmcts.cmc.claimstore.controllers;
+
+import org.junit.Test;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.cmc.claimstore.BaseSaveTest;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimData;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaimData;
+import uk.gov.hmcts.document.utils.InMemoryMultipartFile;
+
+import java.util.Optional;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.successfulDocumentManagementUploadResponse;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.unsuccessfulDocumentManagementUploadResponse;
+
+@TestPropertySource(
+    properties = {
+        "feature_toggles.document_management=true"
+    }
+)
+public class SaveClaimWithDocumentManagementTest extends BaseSaveTest {
+
+    @Test
+    public void shouldUploadSealedCopyOfNonRepresentedClaimIntoDocumentManagementStore() throws Exception {
+        assertSealedClaimIsUploadedIntoDocumentManagementStore(SampleClaimData.submittedByClaimant());
+    }
+
+    @Test
+    public void shouldUploadSealedCopyOfRepresentedClaimIntoDocumentManagementStore() throws Exception {
+        assertSealedClaimIsUploadedIntoDocumentManagementStore(SampleClaimData.submittedByLegalRepresentative());
+    }
+
+    private void assertSealedClaimIsUploadedIntoDocumentManagementStore(ClaimData claimData) throws Exception {
+        given(documentUploadClient.upload(eq(AUTHORISATION_TOKEN), anyList()))
+            .willReturn(successfulDocumentManagementUploadResponse());
+
+        MvcResult result = makeRequest(claimData)
+            .andExpect(status().isOk())
+            .andReturn();
+
+        verify(documentUploadClient).upload(AUTHORISATION_TOKEN, newArrayList(new InMemoryMultipartFile("files",
+            deserializeObjectFrom(result, Claim.class).getReferenceNumber() + "-sealed-claim.pdf",
+            "application/pdf", new byte[]{1, 2, 3, 4})
```

Comment:
`new byte[]{1, 2, 3, 4}` literal is repeated a number of times, can we extract a constant?

Reply:
Done 👍